parser/proto.go

```diff
@@ -47,120 +49,162 @@ func (p *ProtoParser) Parse(bs []byte) (model.Metric, error) {
 
 type ProtoMetric struct {
 	pp  *Pool
-	msg *dynamic.Message
+	msg protoreflect.Message
 }
 
 func (m *ProtoMetric) GetBool(key string, nullable bool) interface{} {
-	value, _ := m.msg.TryGetFieldByName(key)
-	if value == nil {
+	field := m.msg.Descriptor().Fields().ByName(protoreflect.Name(key))
+	if field == nil {
 		return getDefaultBool(nullable)
 	}
 
-	return getBool(reflect.ValueOf(value), nullable)
+	return getBool(field, m.msg.Get(field), nullable)
 }
 
 func (m *ProtoMetric) GetInt8(key string, nullable bool) interface{} {
-	value, _ := m.msg.TryGetFieldByName(key)
-	return getIntFromProto[int8](value, nullable, math.MinInt8, math.MaxInt8)
+	field := m.msg.Descriptor().Fields().ByName(protoreflect.Name(key))
+	return getIntFromProto[int8](field, m.msg.Get(field), nullable, math.MinInt8, math.MaxInt8)
```

Comment:
Maybe you was right, that we should check field for nil before (outside of the `getIntFromProto`), because m.msg.Get can panics if field is nil. But need to check.